starfish/imagestack/test/test_imagestack_index.py

```diff
@@ -66,3 +68,107 @@ def test_imagestack_indexing():
     expected_shape = OrderedDict([(Axes.ROUND, 1), (Axes.CH, 1),
                                   (Axes.ZPLANE, 1), (Axes.Y, 190), (Axes.X, 190)])
     assert indexed_stack.shape == expected_shape
+
+
+X_COORDS = 1, 2
+Y_COORDS = 4, 6
+Z_COORDS = 1, 3
+
+
+def test_find_nearest():
+    """
+    Set up ImageStack with physical coordinates:
+        x_coords = [1. 1.11111111 1.22222222 1.33333333 1.44444444 1.55555556
+        1.66666667 1.77777778 1.88888889 2.]
+
+        y_coords = [4. 4.22222222 4.44444444 4.66666667 4.88888889 5.11111111
+        5.33333333 5.55555556 5.77777778 6. ]
+
+     Test that find_nearest() finds the correct corresponding positional index values
+    """
+    stack_shape = OrderedDict([(Axes.ROUND, 3), (Axes.CH, 2),
+                               (Axes.ZPLANE, 1), (Axes.Y, 10), (Axes.X, 10)])
+
+    physical_coords = OrderedDict([(PhysicalCoordinateTypes.X_MIN, X_COORDS[0]),
+                                   (PhysicalCoordinateTypes.X_MAX, X_COORDS[1]),
+                                   (PhysicalCoordinateTypes.Y_MIN, Y_COORDS[0]),
+                                   (PhysicalCoordinateTypes.Y_MAX, Y_COORDS[1]),
+                                   (PhysicalCoordinateTypes.Z_MIN, Z_COORDS[0]),
+                                   (PhysicalCoordinateTypes.Z_MAX, Z_COORDS[1])])
+
+    stack = test_utils.imagestack_with_coords_factory(stack_shape, physical_coords)
+    assert iu.find_nearest(stack.xarray[Coordinates.X.value], 1.2) == 2
+    assert iu.find_nearest(stack.xarray[Coordinates.X.value], 1.5) == 4
+    assert iu.find_nearest(stack.xarray[Coordinates.X.value], (1.2, 1.5)) == (2, 4)
+
+    assert iu.find_nearest(stack.xarray[Coordinates.Y.value], 4) == 0
+    assert iu.find_nearest(stack.xarray[Coordinates.Y.value], 5.1) == 5
+    assert iu.find_nearest(stack.xarray[Coordinates.Y.value], (4, 5.1)) == (0, 5)
+
+    # assert values outside the range are given min/max of array
+    assert iu.find_nearest(stack.xarray[Coordinates.X.value], 5) == 9
+    assert iu.find_nearest(stack.xarray[Coordinates.X.value], -5) == 0
+
+
+def test_convert_coords_to_indices():
```

Comment:
can we add a test that does both coordinate and index selection?

Reply:
added!